src/cache/impl/LevelDbCache.ts

```diff
@@ -94,21 +94,38 @@ export class LevelDbCache<V = any> implements SortKeyCache<V> {
   }
 
   async put(stateCacheKey: CacheKey, value: V): Promise<void> {
-    const contractCache = this.db.sublevel<string, any>(stateCacheKey.contractTxId, { valueEncoding: 'json' });
+    const contractCache = this.db.sublevel<string, any>(stateCacheKey.key, { valueEncoding: 'json' });
     // manually opening to fix https://github.com/Level/level/issues/221
     await contractCache.open();
     await contractCache.put(stateCacheKey.sortKey, value);
   }
 
-  async delete(contractTxId: string): Promise<void> {
-    const contractCache = this.db.sublevel<string, any>(contractTxId, { valueEncoding: 'json' });
+  async delete(key: string): Promise<void> {
+    const contractCache = this.db.sublevel<string, any>(key, { valueEncoding: 'json' });
     await contractCache.open();
     await contractCache.clear();
   }
 
-  close(): Promise<void> {
-    return null;
-    //return this.db.close();
+  async batch(opStack: BatchDBOp<V>[]) {
+    for (const op of opStack) {
```

Comment:
[db.batch](https://github.com/Level/level#dbbatchoperations-options-callback) ?

Reply:
Yes, that's a good point, but there is a small issue with how we store values in sublevels.

For example we use transaction ids to get the sublevels and then sortKeys in sublevel.
If we would swich the transactionId with the sortKey the batch function would fit very well.

Reply:
yeah, fck it for now.